akka-actor/src/main/scala/akka/event/Logging.scala

```diff
@@ -1233,9 +1298,84 @@ object NoLogging extends LoggingAdapter {
   final override def isInfoEnabled = false
   final override def isDebugEnabled = false
 
-  final protected override def notifyError(message: String): Unit = ()
-  final protected override def notifyError(cause: Throwable, message: String): Unit = ()
-  final protected override def notifyWarning(message: String): Unit = ()
-  final protected override def notifyInfo(message: String): Unit = ()
-  final protected override def notifyDebug(message: String): Unit = ()
+  final protected[akka] override def notifyError(message: String): Unit = ()
+  final protected[akka] override def notifyError(cause: Throwable, message: String): Unit = ()
+  final protected[akka] override def notifyWarning(message: String): Unit = ()
+  final protected[akka] override def notifyInfo(message: String): Unit = ()
+  final protected[akka] override def notifyDebug(message: String): Unit = ()
+}
+
+class LoggingAdapterExt(delegate: LoggingAdapter) extends LoggingAdapter {
```

Comment:
I tried that approach. Getting compiler error on `@annotation.varargs`:

```
[error] (akka-actor/compile:compile) java.lang.AssertionError: assertion failed: no extension method found for:
[error]
[error]   method debug:(template: String, args: Any*)Unit
[error]
[error]  Candidates:
[error]
[error]
[error]
[error]  Candidates (signatures normalized):
[error]
[error]
[error]
[error]  Eligible Names: debug$extension0,debug$extension1"
[error] Total time: 2 s, completed Nov 11, 2015 11:37:17 AM
```

Without the annotation Java code does not work.

![screen shot 2015-11-11 at 11 40 25 am](https://cloud.githubusercontent.com/assets/1329454/11100953/129418d6-8869-11e5-96d6-1a9a619ac637.png)

And Java code looks ugly (I added implicit class to a companion object for LoggingAdapter):

```
final LoggingAdapter log = Logging.getLogger(system, this);
final LoggingAdapter.LoggingAdapterExt logext = new LoggingAdapter.LoggingAdapterExt(log);
```

I think I'm hitting a threshold on that ticket, in terms of value/effort. Maybe we better put it into a backlog until we have a release where we can add backward incompatible changes and just add the method into LoggingAdapter as suggested initially?